addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/cache/ExpiringCacheAsync.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.cache;
```

Comment:
I was thinking about moving it to a separate bundle as it turns out to be a quite handy class for my bindings. It is actually always the case, that we don't want to block in `handleCommand` and return a cached value but at the same time retrieve a more recent one if the cached one is too old.